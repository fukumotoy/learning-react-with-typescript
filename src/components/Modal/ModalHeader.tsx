import React, { ElementType, FC, HTMLProps, useContext } from 'react';
import classnames from 'classnames';
import ModalContext from './ModalContext';

interface Props extends HTMLProps<HTMLElement> {
  asElement?: ElementType;
}

const ModalHeader: FC<Props> = ({
  asElement: Component = 'div',
  children,
  className,
  ...props
}: Props): JSX.Element => {
  const { dismiss } = useContext(ModalContext);

  return (
    <Component {...props} className={classnames('Modal__header', className)}>
      {children}

      <button
        type="button"
        className="Modal__dismiss"
        aria-label="閉じる"
        onClick={dismiss}
      >
        <span aria-hidden="true">&times;</span>
      </button>
    </Component>
  );
};

export default ModalHeader;
