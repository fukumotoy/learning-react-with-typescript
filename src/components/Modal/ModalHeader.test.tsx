/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ModalHeader from './ModalHeader';

it('renders correctly', (): void => {
  const tree = renderer.create(<ModalHeader>Text</ModalHeader>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text', (): void => {
  const text = 'Text';
  const mock = shallow(<ModalHeader>{text}</ModalHeader>);

  expect(mock.text()).toMatch(text);
});

it('replace tag', (): void => {
  const tag = 'header';
  const mock = shallow(<ModalHeader asElement={tag}>Text</ModalHeader>);
  expect(mock.name()).toBe(tag);
});
