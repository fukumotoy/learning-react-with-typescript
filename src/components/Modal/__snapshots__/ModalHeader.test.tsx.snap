// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Modal__header"
>
  Text
  <button
    aria-label="閉じる"
    className="Modal__dismiss"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;
